frontend/src/shared/video-card-image.tsx

```diff
@@ -0,0 +1,24 @@
+"use client";
+
+interface Props {
+  videoId: string;
+  progress?: number;
+}
+
+/**
+ * Renders the thumbnail of a video or the progress of the video if it is being processed
+ */
+export default function VideoCardImage({ progress, videoId }: Props) {
+  if (!progress) {
+    return <img src={`/api/clips/${videoId}/thumbnail.jpg`} />
+  }
+
+  if (progress === -1) {
+    return <div style={{ userSelect: "none" }}>Queued</div>
+  }
+
+  // @TODO: Since we have a thumbnail here we can maybe add some opacity to it and show the progress on top of it
+  return (
+    <div className="radial-progress" style={{"--value":progress} as any}>{`${progress}%`}</div>
```

Comment:
In the case that we do reduce the frequency we're requesting progress of videos, is there any good way to make a smooth animation for movement of the progress bar? So it doesn't look so jumpy?